detectable/src/main/java/com/synopsys/integration/detectable/detectable/util/DependencyTypeFilter.java

```diff
@@ -0,0 +1,30 @@
+package com.synopsys.integration.detectable.detectable.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.jetbrains.annotations.Nullable;
+
+public class DependencyTypeFilter<T extends Enum<T>> {
+    private final List<T> allowedDependencyTypes;
+
+    @SafeVarargs
+    public DependencyTypeFilter(T... allowedDependencyTypes) {
```

Comment:
I added this constructor to help with tests, but it allows you to easily construct an empty `DependencyTypeFilter`
which is not a use case for application code. Thoughts?